### Import Library

In [5]:
import wikipedia
import nltk
#nltk.download()

### Web Scrapping using wikipedia

In [6]:
wiki_data=wikipedia.page("Machinelearning")
wiki_data.content

'Machine learning (ML) is a field of inquiry devoted to understanding and building methods that \'learn\', that is, methods that leverage data to improve performance on some set of tasks. It is seen as a part of artificial intelligence. \nMachine learning algorithms build a model based on sample data, known as training data, in order to make predictions or decisions without being explicitly programmed to do so. Machine learning algorithms are used in a wide variety of applications, such as in medicine, email filtering, speech recognition, agriculture, and computer vision, where it is difficult or unfeasible to develop conventional algorithms to perform the needed tasks.A subset of machine learning is closely related to computational statistics, which focuses on making predictions using computers, but not all machine learning is statistical learning. The study of mathematical optimization delivers methods, theory and application domains to the field of machine learning. Data mining is a

### Images in Wikipedia content

In [7]:
wiki_data.images

['https://upload.wikimedia.org/wikipedia/commons/1/1b/AI_hierarchy.svg',
 'https://upload.wikimedia.org/wikipedia/commons/7/7a/Anatomy-1751201_1280.png',
 'https://upload.wikimedia.org/wikipedia/commons/4/46/Colored_neural_network.svg',
 'https://upload.wikimedia.org/wikipedia/commons/5/51/Computer_Retro.svg',
 'https://upload.wikimedia.org/wikipedia/commons/e/eb/Decision_Tree.jpg',
 'https://upload.wikimedia.org/wikipedia/commons/2/23/Fig-y_Part_of_ML_as_subfield_of_AI_or_AI_as_subfield_of_ML.jpg',
 'https://upload.wikimedia.org/wikipedia/commons/4/41/Global_thinking.svg',
 'https://upload.wikimedia.org/wikipedia/commons/f/fe/Kernel_Machine.svg',
 'https://upload.wikimedia.org/wikipedia/commons/3/3a/Linear_regression.svg',
 'https://upload.wikimedia.org/wikipedia/commons/6/68/Overfitted_Data.png',
 'https://upload.wikimedia.org/wikipedia/commons/3/37/People_icon.svg',
 'https://upload.wikimedia.org/wikipedia/commons/a/a7/Regressions_sine_demo.svg',
 'https://upload.wikimedia.org/wikip

### Create Different variable for further process

In [8]:
data=wiki_data

### Creating Stem of a word.

In [9]:
from nltk.stem import PorterStemmer

In [15]:
pt=PorterStemmer()
list_words=['studies','Study','rejoin','Hungry','carefully']
for i in list_words:
    s=pt.stem(i)
    print(f"Original word is : {i} \nIts Stem is : {s}")
    print()

Original word is : studies 
Its Stem is : studi

Original word is : Study 
Its Stem is : studi

Original word is : rejoin 
Its Stem is : rejoin

Original word is : Hungry 
Its Stem is : hungri

Original word is : carefully 
Its Stem is : care



### Creating Lemma

In [16]:
from nltk.stem import WordNetLemmatizer

In [19]:
wl=WordNetLemmatizer()
list_words=['studies','Study','rejoin','Hungry','carefully']
for i in list_words:
    l=wl.lemmatize(i)
    print(f"Original word is : {i}, \nIts Lemma is {l}")
    print()

Original word is : studies, 
Its Lemma is study

Original word is : Study, 
Its Lemma is Study

Original word is : rejoin, 
Its Lemma is rejoin

Original word is : Hungry, 
Its Lemma is Hungry

Original word is : carefully, 
Its Lemma is carefully



#### Stem & Lemma

In [20]:
list_words=['studies','Study','rejoin','Hungry','carefully']
wl=WordNetLemmatizer()
pt=PorterStemmer()
for i in list_words:
    l=wl.lemmatize(i)
    s=pt.stem(i)
    print(f"Original word is : {i} \nIts Stem is : {s}, Lemma is : {l}")
    print()

Original word is : studies 
Its Stem is : studi, Lemma is : study

Original word is : Study 
Its Stem is : studi, Lemma is : Study

Original word is : rejoin 
Its Stem is : rejoin, Lemma is : rejoin

Original word is : Hungry 
Its Stem is : hungri, Lemma is : Hungry

Original word is : carefully 
Its Stem is : care, Lemma is : carefully



In [24]:
wiki_words=[]
lemma_words=[]
stem_words=[]
for i in str(data).split:
    wiki_words.append(i)
    lemma_words.append(wl.lemmatize(i))
    stem_words.append(pt.stem(i))

TypeError: 'builtin_function_or_method' object is not iterable

In [19]:
len(set(wiki_words))

0

In [20]:
len(set(lemma_words))

0

In [21]:
len(set(stem_words))

0

### Stopwords

In [26]:
from nltk.corpus import stopwords

In [30]:
my_stopwords=stopwords.words()
len(my_stopwords)

10405

**In all languages there are 10405 StopWords**

### Check Stopwords in every language

In [31]:
lan_stopwords=stopwords.fileids()

In [32]:
for i in lan_stopwords:
    words=len(stopwords.words(i))
    print(f"Language {i}, Stopwords {words}")

Language arabic, Stopwords 754
Language azerbaijani, Stopwords 165
Language basque, Stopwords 326
Language bengali, Stopwords 398
Language catalan, Stopwords 278
Language chinese, Stopwords 841
Language danish, Stopwords 94
Language dutch, Stopwords 101
Language english, Stopwords 179
Language finnish, Stopwords 235
Language french, Stopwords 157
Language german, Stopwords 232
Language greek, Stopwords 265
Language hebrew, Stopwords 221
Language hinglish, Stopwords 1036
Language hungarian, Stopwords 199
Language indonesian, Stopwords 758
Language italian, Stopwords 279
Language kazakh, Stopwords 324
Language nepali, Stopwords 255
Language norwegian, Stopwords 176
Language portuguese, Stopwords 207
Language romanian, Stopwords 356
Language russian, Stopwords 151
Language slovene, Stopwords 1784
Language spanish, Stopwords 313
Language swedish, Stopwords 114
Language tajik, Stopwords 163
Language turkish, Stopwords 53


In [33]:
sent="I do not like to go out for shopping"
for i in sent.split():
    if i.lower not in my_stopwords:
        print(i)

I
do
not
like
to
go
out
for
shopping
